### PR TITLE
new(xychart): add Axes + Grid components

### PR DESCRIPTION
**Pull request by williaster**
TODO
- [x] update base branch to `master` when #789 lands (this work builds on it)

#### :rocket: Enhancements

This PR 
- adds `Axis`, `AnimatedAxis`, `Grid`, and `AnimatedGrid` to `@vx/xychart`
  - handful of misc fixes which became apparent when adding these components
- adds tests for all of these components

#### 🐛 Bug fixes

[`react-spring`]
- fixes issue where `top/left` were not passed properly to `AnimatedGridRows/Columns`

Here's a demo of it (demo changes coming in another PR)

@kristw @hshoff @techniq

### CODE REVIEW

File: packages/vx-react-spring/src/axis/AnimatedAxis.tsx [hunk 1]
```diff
@@ -2,13 +2,17 @@ import React, { useMemo } from 'react';
 import Axis, { AxisProps } from '@vx/axis/lib/axis/Axis';
 import { AxisScale, TicksRendererProps } from '@vx/axis/lib/types';
 import AnimatedTicks from './AnimatedTicks';
-import { TransitionConfig } from '../spring-configs/useLineTransitionConfig';
+import { AnimationTrajectory } from '../types';
+
+export type AnimatedAxisProps<Scale extends AxisScale> = Omit<
+  AxisProps<Scale>,
+  'ticksComponent'
+> & { animationTrajectory?: AnimationTrajectory };
```
Comment (williaster) [inline packages/vx-react-spring/src/axis/AnimatedAxis.tsx:10]: this is just a simpler way to add the `AnimationTrajectory` prop

File: packages/vx-xychart/src/components/axis/BaseAxis.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  const tickLabelProps = useMemo(
+    () =>
+      props.tickLabelProps || // use props if passed
+      (axisStyles // else construct from theme if possible
+        ? () =>
+            orientation === 'left' || orientation === 'right'
+              ? // by default, wrap vertical-axis tick labels within the allotted margin space
+                // this does not currently account for axis label
+                { ...axisStyles.tickLabel, width: margin?.[orientation] }
```
Comment (williaster) [inline packages/vx-xychart/src/components/axis/BaseAxis.tsx:44]: this logic could prob be made smarter in the future, and extended to `labelProps`, but didn't implement for now.

File: packages/vx-xychart/src/components/grid/AnimatedGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import { CommonGridProps } from '@vx/grid/lib/types';
+import DataContext from '../../context/DataContext';
+
+export type AnimatedGridProps = {
+  rows?: boolean;
+  columns?: boolean;
+  animationTrajectory?: AnimationTrajectory;
+} & CommonGridProps;
+
+export default function AnimatedGrid({ rows = true, columns = true, ...props }: AnimatedGridProps) {
```
Comment (williaster) [inline packages/vx-xychart/src/components/grid/AnimatedGrid.tsx:14]: might be able to pull out a `BaseGrid` to get rid of the common boilerplate between `Grid` + `AnimatedGrid`

File: packages/vx-xychart/src/components/axis/BaseAxis.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      ? margin?.top ?? 0
+      : 0;
+  const leftOffset =
+    orientation === 'left'
+      ? margin?.left ?? 0
+      : orientation === 'right'
+      ? (width ?? 0) - (margin?.right ?? 0)
+      : 0;
+
+  console.log('baseaxis', { xScale, yScale });
```
Comment (kristw) [inline packages/vx-xychart/src/components/axis/BaseAxis.tsx:64]: remove `console.log`

**Review by kristw — APPROVED** [review]

**Pull request merged**